pkg/server/router/model.go

```diff
@@ -150,6 +150,13 @@ func (p *Payload) HasMasterKey() bool {
 	return p.AccessKey == MasterAccessKey
 }
 
+// IsValidPluginRequest returns true when the payload indicate that the
+// request is from plugin AND the master key is used
+func (p Payload) IsValidPluginRequest() bool {
+	fromPlugin, _ := p.Data["_from_plugin"].(bool)
```

Comment:
Especially that the “HasMasterKey” function only returns valid value after a auth preprocessor update it.